list.cpp

```diff
@@ -0,0 +1,145 @@
+#include "list.h"
+#include <iostream>
+using std::cout;
+using std::endl;
+
+
+List::List(){
+	head = nullptr;
+	node = nullptr;
+	curr1 = nullptr;
+	curr2 = nullptr;
+}
+
+void List::addNode(int idata, int option){
+	node = new Node;	//create a node using dynamic memory allocation
+	node->data = idata;
+
+	//check if the node just created is a first node
+	if (head == nullptr){
+		head = node;
+		curr1 = node;
+		curr2 = node;
+		node->next = nullptr;
+	}
+	else{
+		//This means if we already have first node then we'll do something else
+
+		//This will add the newly created node to the end of the list
+		if (option == 1){
+			curr1 = head;
+
+			while (curr1->next != nullptr){
+
+				curr1 = curr1->next;
+			}//end while
+
+			curr1->next = node;
+			node->next = nullptr;
+
+
+		}//end option 1
+
+		//This will add the newly created node to the beginning of the list
+		if (option == 2){
+
+			curr1 = head;
+			head = node;
+			node->next = curr1;
+		}//end option 2
+
+		//This will add the newly created node to somewhere in the list
+		if (option == 3){
+			
+			curr1 = curr2 = head;
+
+			while (curr2->data != idata){
+				curr1 = curr2;
+				curr2 = curr2->next;
+
+				//if curr2 reaches the end of the list or data couldn't be found (typo?)
+				if (curr2->next == nullptr || curr2->data != idata){
+					cout << "Cannot find the requested data" << endl;
+				}
```

Comment:
if i'm reading this correctly, this may happen many times before the data is, in fact, found in the list -- did you mean `&&`?

also, if the data is never found, how does it break out of the `while` loop?